api/resolvers/data/student.ts

```diff
@@ -207,4 +223,74 @@ export class StudentResolver {
 
     return await StudentDropoutDataLoader.load(id);
   }
+
+  @FieldResolver()
+  async admission(
+    @Root() { id }: PartialStudent
+  ): Promise<Admission | undefined> {
+    assertIsDefined(
+      id,
+      `student id needs to be available for Student field resolvers`
+    );
+
+    return await StudentAdmissionDataLoader.load(id);
+  }
+
+  @FieldResolver()
+  async employed(
+    @Root() { id }: PartialStudent
+  ): Promise<Employed | undefined> {
+    assertIsDefined(
+      id,
+      `student id needs to be available for Student field resolvers`
+    );
+
+    return await StudentEmployedDataLoader.load(id);
+  }
+
+  @FieldResolver()
+  async n_cycles(
+    @Root() { programs, curriculums }: PartialStudent
+  ): Promise<$PropertyType<Student, "n_cycles">> {
+    const total_cycles = await StudentListCyclesDataLoader.load({
+      program_id: programs![0]["id"],
+      curriculum: curriculums!,
+    });
+    const list_cycle = total_cycles.map((d) => d.course_cat);
+
+    return list_cycle;
+  }
+  @FieldResolver()
+  async n_courses_cycles(
+    @Root() { programs, curriculums, id }: PartialStudent
+  ): Promise<$PropertyType<Student, "n_courses_cycles">> {
+    const total_cycles = await StudentListCyclesDataLoader.load({
+      program_id: programs![0]["id"],
+      curriculum: curriculums!,
+    });
+
+    const list_cycle = total_cycles.map((d) => d.course_cat);
+    var valores_totales = [];
+
+    for await (let cycle of list_cycle) {
```

Comment:
este "for await" no está haciendo nada, list_cycle no es ni un interador asyncrono ni generador

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of

Reply:
solucionado